azerothcore/azerothcore-wotlk
# BG Queue Announcer FIX

##### CHANGES PROPOSED:

-  Show to the player or to the world ( config based ) the queue status for the current bg


###### ISSUES ADDRESSED:


Closes 


##### TESTS PERFORMED:




##### HOW TO TEST THE CHANGES:

Go into worldserver.conf set `Battleground.QueueAnnouncer.Enable` to 1 and `Battleground.QueueAnnouncer.PlayerOnly ` first time to 0 and then to 1.
To test the changes queue to a bg ( not random ) and spam the button to join the queue.
Try it again in 30 seconds or spam it for 30+ seconds ( so it will write again and you can see that the spam protection works )
```
#
#    Battleground.QueueAnnouncer.Enable
#        Description: Announce battleground queue status to chat.
#        Default:     0 - (Disabled)
#                     1 - (Enabled)

Battleground.QueueAnnouncer.Enable = 1

#
#    Battleground.QueueAnnouncer.PlayerOnly
#        Description: Battleground queue announcement type.
#        Default:     0 - (System message, Anyone can see it)
#                     1 - (Private, Only queued players can see it)

Battleground.QueueAnnouncer.PlayerOnly = 0
```


##### KNOWN ISSUES AND TODO LIST:


- [x] It can be spammed at the moment


##### Target branch(es):

Master
